pkg/provisioner/crdprovisioner.go

```diff
@@ -327,15 +346,12 @@ func (c *CrdProvisioner) PublishVolume(
 	azVolumeAttachmentInstance, err := azVA.Get(ctx, attachmentName, metav1.GetOptions{})
 	if err == nil {
 		updated := azVolumeAttachmentInstance.DeepCopy()
-		if updated.Status != nil && updated.Status.Error != nil {
-			updated.Status = nil
-			updated, err = azVA.UpdateStatus(ctx, updated, metav1.UpdateOptions{})
-			if err != nil || updated == nil || updated.Status != nil {
-				klog.Errorf("Failed to update azVolumeAttachment resource status for volume name (%s), error: %v", volumeName, err)
-				return nil, err
-			}
-		}
 
+		// reset error and state field of the AzVolumeAttachment so that the reconciler can retry attachment
+		if updated.Status.Error != nil {
+			updated.Status.Error = nil
+			updated.Status.State = v1alpha1.AttachmentPending
+		}
```

Comment:
Same as above. azVA.UpdateStatus() after this?

Reply:
In this case, we wouldn't need a separate update call as we already have a azVA.Update call in line 360.

Reply:
The reason we had a separate updateStatus call previously was that we defined status as a subresource but now that it no longer is, a regular update call should update the status as well.

Reply:
Why is `Status` no longer a sub-resource? What is the impact of making that change?

Reply:
`Status` is no longer a subresource because with the modification we made to the status struct, it no longer made sense for it to be a pointer variable. And leaving status as a subresource was causing issues with the creation of `AzVolume` and `AzVolumeAttachment` object.

The impact of this change would essentially boil down to what change each `update` call would be making to our CRIs. Previously, it only updated the spec and metadata fields but now it would be updating the status field as well. So we should take this into account when updating objects.